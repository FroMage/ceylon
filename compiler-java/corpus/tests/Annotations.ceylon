class Annotations() {

	class NoArguments() {

		public deprecated default
		void methodAnnotations(deprecated Natural param=0) {}
		
		public deprecated default
		Natural attributeAnnotations = 1;
		
		package deprecated abstract
		class ClassAnnotations() {}
		
		module
		interface InterfaceAnnotations {}
		
	}

	class LiteralArguments() {	

		doc "The method doc"
		by "Gavin King"
		   "Andrew Haley"
		version '1.2.0'
		timeout 1000
		see #ClassAnnotations
		    #methodAnnotations
		    #Object.equals
		throws #DatabaseException
	           "if database access fails"
		void methodAnnotations(
			deprecated
			doc "use something else" 
			see #otherMethod 
			Natural param=0) {}
		
		doc "The attribute doc"
		by "Gavin King"
		   "Andrew Haley"
		see #ClassAnnotations
		    #methodAnnotations
		    #Object.equals
		version '1.0.0'
		Natural attributeAnnotations = 1;
		
		see #ClassAnnotations
		    #methodAnnotations
		    #Object.equals
		doc "The class doc"
		throws #DatabaseException
	           "if database access fails"
		version '2.1.0'
		by "Gavin King"
		   "Andrew Haley"
		class ClassAnnotations() {}
		
		doc "The interface doc"
		version '1.0.0 beta'
		see #ClassAnnotations
		    #methodAnnotations
		    #Object.equals
		by "Gavin King"
		   "Andrew Haley"
		interface InterfaceAnnotations {}
		
	}

	class NamedArguments() {

	    doc { text="some text about the method"; }
		lock { timeout=1000; }
		throws { type = #DatabaseException;
	             description = "if database access fails"; }
		by { "Gavin King", "Andrew Haley" }
		see { #ClassAnnotations, #methodAnnotations, #Object.equals }
		void methodAnnotations(
			deprecated
			doc { text="use something else"; }
			see { #otherMethod }
			Natural param=0) {}
		
		see { #ClassAnnotations, #methodAnnotations, #Object.equals }
		doc { text="some text about the attribute"; }
		by { "Gavin King" }
		column { name="columnName"; comment="some comment"; }
		Natural attributeAnnotations = 1;
		
		by { "Gavin King", "Andrew Haley" }
		doc { text="some text about the class"; }
		throws { type = #DatabaseException;
	             description = "if database access fails"; }
		table { name="tableName"; schema="schemaName"; }
		see { #ClassAnnotations, #methodAnnotations, #Object.equals }
		scope { scopeType=session; }
		class ClassAnnotations() {}
		
		doc { text="some text about the interface"; }
		see { #ClassAnnotations, #methodAnnotations, #Object.equals }
		by { "Andrew Haley" }
		interface InterfaceAnnotations {}
	
	}
	
	class PositionalArguments() {
	
		throws(#DatabaseException,
	           "if database access fails")
	    doc("some text about the method")
		lock(1000)
		version('2.1.0')
		by("Gavin King", "Andrew Haley")
		see (#ClassAnnotations, #methodAnnotations, #Object.equals)
		void methodAnnotations(
			deprecated
			doc("use something else") 
			see(#otherMethod) 
			Natural param=0) {}
		
		doc("some text about the attribute")
		by("Gavin King", "Andrew Haley")
		see (#ClassAnnotations, #methodAnnotations, #Object.equals)
		version('1.2.0')
		Natural attributeAnnotations = 1;
		
		version('1.0.0')
		by("Gavin King", "Andrew Haley")
		see (#ClassAnnotations, #methodAnnotations, #Object.equals)
		doc("some text about the class")
		scope(session)
		throws(#DatabaseException,
	           "if database access fails")
		class ClassAnnotations() {}
		
		see (#ClassAnnotations, #methodAnnotations, #Object.equals)
		version('1.0.0 beta')
		doc("some text about the interface")
		by("Gavin King", "Andrew Haley")
		interface InterfaceAnnotations {}
	
	}
	
	class ComplexArguments() {
	    
	    queries( Query('select p.name from Person p'), 
	             Query('select p.name from Person p where p.age>18') )
	    see #List<String>
	    class ClassAnnotations() {}
	
	}
	
}