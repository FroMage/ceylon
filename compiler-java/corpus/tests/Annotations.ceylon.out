# Annotations.ceylon

(TYPE_DECL
  (TYPE_DECL
    (CLASS_DECL
      (TYPE_NAME Annotations)
      (STMT_LIST
        (TYPE_DECL
          (CLASS_DECL
            (TYPE_NAME NoArguments)
            (STMT_LIST
              (MEMBER_DECL
                (MEMBER_TYPE void)
                (MEMBER_NAME methodAnnotations)
                (FORMAL_PARAMETER_LIST
                  (FORMAL_PARAMETER
                    (ANNOTATION_LIST
                      (ANNOTATION
                        (ANNOTATION_NAME deprecated)))
                    (TYPE
                      (TYPE_NAME Natural))
                    (ARG_NAME param)
                    (INIT_EXPR
                      (EXPR
                        (INT_CST 0))))) STMT_LIST
                (ANNOTATION_LIST
                  (ANNOTATION PUBLIC)
                  (ANNOTATION
                    (ANNOTATION_NAME deprecated))
                  (ANNOTATION default)))
              (MEMBER_DECL
                (MEMBER_TYPE
                  (TYPE
                    (TYPE_NAME Natural)))
                (MEMBER_NAME attributeAnnotations)
                (INIT_EXPR
                  (EXPR
                    (INT_CST 1)))
                (ANNOTATION_LIST
                  (ANNOTATION PUBLIC)
                  (ANNOTATION
                    (ANNOTATION_NAME deprecated))
                  (ANNOTATION default)))
              (TYPE_DECL
                (CLASS_DECL
                  (TYPE_NAME ClassAnnotations) STMT_LIST)
                (ANNOTATION_LIST
                  (ANNOTATION package)
                  (ANNOTATION
                    (ANNOTATION_NAME deprecated))
                  (ANNOTATION abstract)))
              (TYPE_DECL
                (INTERFACE_DECL
                  (TYPE_NAME InterfaceAnnotations))
                (ANNOTATION_LIST
                  (ANNOTATION module))))))
        (TYPE_DECL
          (CLASS_DECL
            (TYPE_NAME LiteralArguments)
            (STMT_LIST
              (MEMBER_DECL
                (MEMBER_TYPE void)
                (MEMBER_NAME methodAnnotations)
                (FORMAL_PARAMETER_LIST
                  (FORMAL_PARAMETER
                    (ANNOTATION_LIST
                      (ANNOTATION
                        (ANNOTATION_NAME deprecated))
                      (ANNOTATION
                        (ANNOTATION_NAME doc)
                        (STRING_CST "use something else"))
                      (ANNOTATION
                        (ANNOTATION_NAME see)))
                    (TYPE
                      (TYPE_NAME Natural))
                    (ARG_NAME param)
                    (INIT_EXPR
                      (EXPR
                        (INT_CST 0))))) STMT_LIST
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (STRING_CST "The method doc"))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (STRING_CST "Andrew Haley"))
                  (ANNOTATION
                    (ANNOTATION_NAME version))
                  (ANNOTATION
                    (ANNOTATION_NAME timeout))
                  (ANNOTATION
                    (ANNOTATION_NAME see))
                  (ANNOTATION
                    (ANNOTATION_NAME throws)
                    (STRING_CST "if database access fails"))))
              (MEMBER_DECL
                (MEMBER_TYPE
                  (TYPE
                    (TYPE_NAME Natural)))
                (MEMBER_NAME attributeAnnotations)
                (INIT_EXPR
                  (EXPR
                    (INT_CST 1)))
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (STRING_CST "The attribute doc"))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (STRING_CST "Andrew Haley"))
                  (ANNOTATION
                    (ANNOTATION_NAME see))
                  (ANNOTATION
                    (ANNOTATION_NAME version))))
              (TYPE_DECL
                (CLASS_DECL
                  (TYPE_NAME ClassAnnotations) STMT_LIST)
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME see))
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (STRING_CST "The class doc"))
                  (ANNOTATION
                    (ANNOTATION_NAME throws)
                    (STRING_CST "if database access fails"))
                  (ANNOTATION
                    (ANNOTATION_NAME version))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (STRING_CST "Andrew Haley"))))
              (TYPE_DECL
                (INTERFACE_DECL
                  (TYPE_NAME InterfaceAnnotations))
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (STRING_CST "The interface doc"))
                  (ANNOTATION
                    (ANNOTATION_NAME version))
                  (ANNOTATION
                    (ANNOTATION_NAME see))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (STRING_CST "Andrew Haley")))))))
        (TYPE_DECL
          (CLASS_DECL
            (TYPE_NAME NamedArguments)
            (STMT_LIST
              (MEMBER_DECL
                (MEMBER_TYPE void)
                (MEMBER_NAME methodAnnotations)
                (FORMAL_PARAMETER_LIST
                  (FORMAL_PARAMETER
                    (ANNOTATION_LIST
                      (ANNOTATION
                        (ANNOTATION_NAME deprecated))
                      (ANNOTATION
                        (ANNOTATION_NAME doc)
                        (ARG_LIST
                          (NAMED_ARG
                            (ARG_NAME text)
                            (INIT_EXPR
                              (EXPR
                                (STRING_CST "use something else"))))))
                      (ANNOTATION
                        (ANNOTATION_NAME see)
                        (ARG_LIST
                          (VARARGS
                            (EXPR_LIST
                              (EXPR
                                (REFLECTED_LITERAL
                                  (MEMBER_NAME otherMethod))))))))
                    (TYPE
                      (TYPE_NAME Natural))
                    (ARG_NAME param)
                    (INIT_EXPR
                      (EXPR
                        (INT_CST 0))))) STMT_LIST
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME text)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "some text about the method"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME lock)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME timeout)
                        (INIT_EXPR
                          (EXPR
                            (INT_CST 1000))))))
                  (ANNOTATION
                    (ANNOTATION_NAME throws)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME type)
                        (INIT_EXPR
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME DatabaseException))))))
                      (NAMED_ARG
                        (ARG_NAME description)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "if database access fails"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (STRING_CST "Gavin King"))
                          (EXPR
                            (STRING_CST "Andrew Haley"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME ClassAnnotations))))
                          (EXPR
                            (REFLECTED_LITERAL
                              (MEMBER_NAME methodAnnotations)))
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME Object))
                              (MEMBER_NAME equals)))))))))
              (MEMBER_DECL
                (MEMBER_TYPE
                  (TYPE
                    (TYPE_NAME Natural)))
                (MEMBER_NAME attributeAnnotations)
                (INIT_EXPR
                  (EXPR
                    (INT_CST 1)))
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME ClassAnnotations))))
                          (EXPR
                            (REFLECTED_LITERAL
                              (MEMBER_NAME methodAnnotations)))
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME Object))
                              (MEMBER_NAME equals)))))))
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME text)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "some text about the attribute"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (STRING_CST "Gavin King"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME column)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME name)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "columnName"))))
                      (NAMED_ARG
                        (ARG_NAME comment)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "some comment"))))))))
              (TYPE_DECL
                (CLASS_DECL
                  (TYPE_NAME ClassAnnotations) STMT_LIST)
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (STRING_CST "Gavin King"))
                          (EXPR
                            (STRING_CST "Andrew Haley"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME text)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "some text about the class"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME throws)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME type)
                        (INIT_EXPR
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME DatabaseException))))))
                      (NAMED_ARG
                        (ARG_NAME description)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "if database access fails"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME table)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME name)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "tableName"))))
                      (NAMED_ARG
                        (ARG_NAME schema)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "schemaName"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME ClassAnnotations))))
                          (EXPR
                            (REFLECTED_LITERAL
                              (MEMBER_NAME methodAnnotations)))
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME Object))
                              (MEMBER_NAME equals)))))))
                  (ANNOTATION
                    (ANNOTATION_NAME scope)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME scopeType)
                        (INIT_EXPR
                          (EXPR
                            (MEMBER_NAME session))))))))
              (TYPE_DECL
                (INTERFACE_DECL
                  (TYPE_NAME InterfaceAnnotations))
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (NAMED_ARG
                        (ARG_NAME text)
                        (INIT_EXPR
                          (EXPR
                            (STRING_CST "some text about the interface"))))))
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME ClassAnnotations))))
                          (EXPR
                            (REFLECTED_LITERAL
                              (MEMBER_NAME methodAnnotations)))
                          (EXPR
                            (REFLECTED_LITERAL
                              (TYPE
                                (TYPE_NAME Object))
                              (MEMBER_NAME equals)))))))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (VARARGS
                        (EXPR_LIST
                          (EXPR
                            (STRING_CST "Andrew Haley")))))))))))
        (TYPE_DECL
          (CLASS_DECL
            (TYPE_NAME PositionalArguments)
            (STMT_LIST
              (MEMBER_DECL
                (MEMBER_TYPE void)
                (MEMBER_NAME methodAnnotations)
                (FORMAL_PARAMETER_LIST
                  (FORMAL_PARAMETER
                    (ANNOTATION_LIST
                      (ANNOTATION
                        (ANNOTATION_NAME deprecated))
                      (ANNOTATION
                        (ANNOTATION_NAME doc)
                        (ARG_LIST
                          (EXPR
                            (STRING_CST "use something else"))))
                      (ANNOTATION
                        (ANNOTATION_NAME see)
                        (ARG_LIST
                          (EXPR
                            (REFLECTED_LITERAL
                              (MEMBER_NAME otherMethod))))))
                    (TYPE
                      (TYPE_NAME Natural))
                    (ARG_NAME param)
                    (INIT_EXPR
                      (EXPR
                        (INT_CST 0))))) STMT_LIST
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME throws)
                    (ARG_LIST
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME DatabaseException))))
                      (EXPR
                        (STRING_CST "if database access fails"))))
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "some text about the method"))))
                  (ANNOTATION
                    (ANNOTATION_NAME lock)
                    (ARG_LIST
                      (EXPR
                        (INT_CST 1000))))
                  (ANNOTATION
                    (ANNOTATION_NAME version)
                    (ARG_LIST
                      (EXPR
                        (QUOTE_CST '2.1.0'))))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "Gavin King"))
                      (EXPR
                        (STRING_CST "Andrew Haley"))))
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME ClassAnnotations))))
                      (EXPR
                        (REFLECTED_LITERAL
                          (MEMBER_NAME methodAnnotations)))
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME Object))
                          (MEMBER_NAME equals)))))))
              (MEMBER_DECL
                (MEMBER_TYPE
                  (TYPE
                    (TYPE_NAME Natural)))
                (MEMBER_NAME attributeAnnotations)
                (INIT_EXPR
                  (EXPR
                    (INT_CST 1)))
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "some text about the attribute"))))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "Gavin King"))
                      (EXPR
                        (STRING_CST "Andrew Haley"))))
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME ClassAnnotations))))
                      (EXPR
                        (REFLECTED_LITERAL
                          (MEMBER_NAME methodAnnotations)))
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME Object))
                          (MEMBER_NAME equals)))))
                  (ANNOTATION
                    (ANNOTATION_NAME version)
                    (ARG_LIST
                      (EXPR
                        (QUOTE_CST '1.2.0'))))))
              (TYPE_DECL
                (CLASS_DECL
                  (TYPE_NAME ClassAnnotations) STMT_LIST)
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME version)
                    (ARG_LIST
                      (EXPR
                        (QUOTE_CST '1.0.0'))))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "Gavin King"))
                      (EXPR
                        (STRING_CST "Andrew Haley"))))
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME ClassAnnotations))))
                      (EXPR
                        (REFLECTED_LITERAL
                          (MEMBER_NAME methodAnnotations)))
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME Object))
                          (MEMBER_NAME equals)))))
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "some text about the class"))))
                  (ANNOTATION
                    (ANNOTATION_NAME scope)
                    (ARG_LIST
                      (EXPR
                        (MEMBER_NAME session))))
                  (ANNOTATION
                    (ANNOTATION_NAME throws)
                    (ARG_LIST
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME DatabaseException))))
                      (EXPR
                        (STRING_CST "if database access fails"))))))
              (TYPE_DECL
                (INTERFACE_DECL
                  (TYPE_NAME InterfaceAnnotations))
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME see)
                    (ARG_LIST
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME ClassAnnotations))))
                      (EXPR
                        (REFLECTED_LITERAL
                          (MEMBER_NAME methodAnnotations)))
                      (EXPR
                        (REFLECTED_LITERAL
                          (TYPE
                            (TYPE_NAME Object))
                          (MEMBER_NAME equals)))))
                  (ANNOTATION
                    (ANNOTATION_NAME version)
                    (ARG_LIST
                      (EXPR
                        (QUOTE_CST '1.0.0 beta'))))
                  (ANNOTATION
                    (ANNOTATION_NAME doc)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "some text about the interface"))))
                  (ANNOTATION
                    (ANNOTATION_NAME by)
                    (ARG_LIST
                      (EXPR
                        (STRING_CST "Gavin King"))
                      (EXPR
                        (STRING_CST "Andrew Haley")))))))))
        (TYPE_DECL
          (CLASS_DECL
            (TYPE_NAME ComplexArguments)
            (STMT_LIST
              (TYPE_DECL
                (CLASS_DECL
                  (TYPE_NAME ClassAnnotations) STMT_LIST)
                (ANNOTATION_LIST
                  (ANNOTATION
                    (ANNOTATION_NAME queries)
                    (ARG_LIST
                      (EXPR
                        (EXPR
                          (TYPE_NAME Query)
                          (CALL_EXPR
                            (ARG_LIST
                              (EXPR
                                (QUOTE_CST 'select p.name from Person p'))))))
                      (EXPR
                        (EXPR
                          (TYPE_NAME Query)
                          (CALL_EXPR
                            (ARG_LIST
                              (EXPR
                                (QUOTE_CST 'select p.name from Person p where p.age>18'))))))))
                  (ANNOTATION
                    (ANNOTATION_NAME see)))))))))))
